### PR TITLE
feat(dedicated): all servers listing page changes

### PR DESCRIPTION
**Pull request by sachinrameshn**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/multi-az` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-12274 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.controller.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        values: reduce(
+          list,
+          (result, item) => ({
+            ...result,
+            [item]: toUpperSnakeCaseFlag
+              ? this.constructor.toUpperSnakeCase(item)
+              : item,
+          }),
+          {},
+        ),
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.controller.js:59]:
```suggestion
        values: list.reduce(
          (result, item) => ({
            ...result,
            [item]: toUpperSnakeCaseFlag
              ? this.constructor.toUpperSnakeCase(item)
              : item,
          }),
          {},
        ),
```

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.controller.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+          [item]: this.$translate.instant(
+            `${translationPrefix}${
+              toUpperSnakeCaseFlag
+                ? this.constructor.toUpperSnakeCase(item)
+                : item
+            }`,
+          ),
+        }),
+        {},
+      ),
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.controller.js:76]:
```suggestion
      values: list.reduce(
        (result, item) => ({
          ...result,
          [item]: this.$translate.instant(
            `${translationPrefix}${
              toUpperSnakeCaseFlag
                ? this.constructor.toUpperSnakeCase(item)
                : item
            }`,
          ),
        }),
        {},
      ),
```

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.controller.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    set(
+      this.ouiDatagridService,
+      'datagrids.dg-servers.paging.offset',
+      currentOffset < this.paginationTotalCount
+        ? currentOffset
+        : this.paginationTotalCount,
+    );
+
+    return this.$q.resolve({
+      data: get(this.dedicatedClusters, 'data'),
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.controller.js:91]:
```suggestion
      data: this.dedicatedClusters?.data,
```

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.controller.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  onPageChange({ pageSize, offset }) {
+    this.onListParamsChange({
+      page: parseInt(offset / pageSize, 10) + 1,
+      pageSize,
+    });
+  }
+
+  onCriteriaChange($criteria) {
+    const filter = $criteria.map((criteria) => ({
+      field: get(criteria, 'property') || 'name',
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.controller.js:107]:
```suggestion
      field: criteria?.property || 'name',
```

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.html [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        <oui-datagrid-column
+            data-title=":: 'dedicated_clusters_name' | translate"
+            data-property="iam.displayName"
+            data-filterable
+            data-sortable
+            data-type="number"
+        >
+            <a data-ng-href="{{ $ctrl.getClusterDashboardLink($row) }}"
+                >{{$row.iam.displayName}}</a
+            >
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.html:21]:
```suggestion
            <a data-ng-href="{{ :: $ctrl.getClusterDashboardLink($row) }}"
                ><span data-ng-bind="$row.iam.displayName"></span></a
            >
```

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.routing.js [hunk 6]
```diff
@@ -25,6 +30,91 @@ export default /* @ngInject */ ($stateProvider) => {
...
+          contains: 'like',
+          is: 'eq',
+          isAfter: 'gt',
+          isBefore: 'lt',
+          isNot: 'ne',
+          smaller: 'lt',
+          bigger: 'gt',
+          startsWith: 'like',
+          endsWith: 'like',
+        };
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.routing.js:66]: Maybe we can reuse directly the constant to layout helper module ?

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/translations/Messages_fr_FR.json [hunk 7]
```diff
@@ -1,4 +1,6 @@
 {
+  "dedicated_servers_all_servers": "All Servers",
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/translations/Messages_fr_FR.json:2]: I think should be translate in French.

Comment (sachinrameshn) [reply packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/servers/translations/Messages_fr_FR.json:2]: Its pending from PO. I have already requested in the ticket.

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.routing.js [hunk 8]
```diff
@@ -25,6 +30,91 @@ export default /* @ngInject */ ($stateProvider) => {
...
+
+        return request.execute(null, true).$promise;
+      },
+      noFiltersServers: /* @ngInject */ (iceberg) =>
+        iceberg('/dedicated/cluster')
+          .query()
+          .expand('CachedObjectList-Pages')
+          .execute(null, true).$promise,
+      paginationNumber: /* @ngInject */ (dedicatedClusters) =>
+        parseInt(get(dedicatedClusters.headers, 'x-pagination-number'), 10),
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.routing.js:95]:
```suggestion
        parseInt(dedicatedClusters.headers['x-pagination-number'], 10),
```

Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.routing.js:95]: Same for other for try to remove get/lodash

File: packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.routing.js (hunk 6 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/clusters.routing.js:66]: with import { ListLayoutHelper } from '@ovh-ux/manager-ng-layout-helpers';
you can use for avoid big diplicate: 

```suggestion
parsedFilters.forEach(({ field, comparator, reference }) => {
            request = request.addFilter(
              field,
              get(ListLayoutHelper.FILTER_OPERATORS, comparator),
              ListLayoutHelper.mapFilterForIceberg(comparator, reference),
            );
          });
          ```

**Review by qpavy — APPROVED** [review]

**Review by ghyenne — APPROVED** [review]

**Pull request merged**